### ISSUE TITLE
I found a bug, I don't know if it is documented

### ISSUE THREAD

**Issue by patrickbreen** [jupyter/notebook#2355]
I found a bug, I don't know if it is documented

If you have a file named `code.py` in the directory that you run `$ jupyter notebook` in, you will get many errors and kernel death, even when opening a new (empty) notebook.

Delete or move the file named `code.py` and the issues magically go away.

**Comment by rgbkrk** [comment]
[`code` is the Interpreter base class from Python](https://docs.python.org/3.5/library/code.html) which seems like an import collision. I can't reproduce this locally with the notebook or `ipython`, so I'm not sure how to make this happen.

**Comment by patrickbreen** [comment]
Ok I figured it out. The problem was that somewhere in the sequence of imports there was an `import pdb` (from the stdlib) which in turn imported:

import os
import re
import sys
import cmd
import bdb
import dis
import code
import glob
import pprint
import signal
import inspect
import traceback
import linecache

And obviously based on the python module lookup system its going to look for `code.py` in `.` before looking in your site packages. So yes I was able to reproduce the same problem if I had `code.py`, `cmd.py`, `dis.py` etc in my working directory.

The moral of the story is: dont name any module in your code the same as a module in the standard lib. A hard thing to do in python with so many standard lib modules.
